src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs

```diff
@@ -144,63 +156,111 @@ public void CanGetFirst()
 
-            SequencePosition c1 = buffer.GetPosition(100);
+            c1 = buffer.GetPosition(0, buffer.Start);
+            Assert.Equal(0, c1.GetInteger());
+            Assert.Equal(bufferSegment3, c1.GetObject());
 
+            c1 = buffer.GetPosition(0, c1);
             Assert.Equal(0, c1.GetInteger());
-            Assert.Equal(bufferSegment4, c1.GetObject());
+            Assert.Equal(bufferSegment3, c1.GetObject());
+
+            c1 = buffer.GetPosition(100);
+            Assert.Equal(0, c1.GetInteger());
+            Assert.Equal(bufferSegment6, c1.GetObject());
 
             c1 = buffer.GetPosition(100, buffer.Start);
+            Assert.Equal(0, c1.GetInteger());
+            Assert.Equal(bufferSegment6, c1.GetObject());
 
+            c1 = buffer.GetPosition(0, c1);
             Assert.Equal(0, c1.GetInteger());
-            Assert.Equal(bufferSegment4, c1.GetObject());
+            Assert.Equal(bufferSegment6, c1.GetObject());
         }
 
         [Fact]
         public void SeekEmptySkipDoesNotCrossPastEnd()
         {
-            var bufferSegment1 = new BufferSegment<byte>(new byte[100]);
-            BufferSegment<byte> bufferSegment2 = bufferSegment1.Append(new byte[0]);
-            BufferSegment<byte> bufferSegment3 = bufferSegment2.Append(new byte[0]);
-            BufferSegment<byte> bufferSegment4 = bufferSegment3.Append(new byte[100]);
+            var bufferSegment1 = new BufferSegment<byte>(new byte[0]);
+            var bufferSegment2 = bufferSegment1.Append(new byte[0]);
+            var bufferSegment3 = bufferSegment2.Append(new byte[100]);
+            var bufferSegment4 = bufferSegment3.Append(new byte[0]);
+            var bufferSegment5 = bufferSegment4.Append(new byte[0]);
+            var bufferSegment6 = bufferSegment5.Append(new byte[100]);
+            var bufferSegment7 = bufferSegment6.Append(new byte[0]);
+            var bufferSegment8 = bufferSegment7.Append(new byte[0]);
+            bufferSegment8.Append(new byte[100]);
+
+            var buffer = new ReadOnlySequence<byte>(bufferSegment1, 0, bufferSegment8, 0);
 
-            var buffer = new ReadOnlySequence<byte>(bufferSegment1, 0, bufferSegment2, 0);
+            SequencePosition c1 = buffer.GetPosition(200);
+            Assert.Equal(0, c1.GetInteger());
+            Assert.Equal(bufferSegment8, c1.GetObject());
 
-            SequencePosition c1 = buffer.GetPosition(100);
+            c1 = buffer.GetPosition(200, buffer.Start);
+            Assert.Equal(0, c1.GetInteger());
+            Assert.Equal(bufferSegment8, c1.GetObject());
 
+            c1 = buffer.GetPosition(0, c1);
             Assert.Equal(0, c1.GetInteger());
-            Assert.Equal(bufferSegment2, c1.GetObject());
+            Assert.Equal(bufferSegment8, c1.GetObject());
 
-            c1 = buffer.GetPosition(100, buffer.Start);
+            c1 = buffer.GetPosition(250);
+            Assert.Equal(0, c1.GetInteger());
```

Comment:
This fails because 250 is 50 bytes into the second array with size; so it should be
```
Assert.Equal(50, c1.GetInteger());
```

Reply:
@benaadams `c1 = buffer.GetPosition(250);` should 
1. return position behind End
2. stop on End and return End
3. Throw some Exception?

What is correct?

Reply:
Ah, see what you are doing...

This might be incorrect behaviour by `ReadOnlySequence`

Reply:
It should throw `ArgumentOutOfRangeException` (and does I believe?)

Reply:
Ah.. looks like there is a corner case not handled

Reply:
Added fix https://github.com/dotnet/corefx/pull/28261

Reply:
I borrowed the tests from https://github.com/dotnet/corefx/pull/28261 that catch this issue in the following PR: https://github.com/dotnet/corefx/pull/28294